core/http/client/src/main/java/org/eclipse/rdf4j/http/client/ModelInterface.java

```diff
@@ -0,0 +1,25 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+
+package org.eclipse.rdf4j.http.client;
+
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.impl.DynamicModelFactory;
+
+public interface ModelInterface {
```

Comment:
This is part of the way the shacl exception is built in the shacl sail. It will be moved into the common module. When that's done we can revisit how it's implemented.

Reply:
I would _really_ prefer keeping this lean, and not introducing additional public interfaces unless there is a clear need for them. Unless I completely misunderstand this does not look like code that an end user should ever want to touch.